### PR TITLE
Added Cypress Test to verify behavior when Guest User is removed from channel

### PR DESCRIPTION
**Pull request by srkgupta**
#### Summary
1. Verify if a Guest user is automatically redirected to another team when removed from all the channels of a team. 
2. Verify if a Guest user is automatically redirected to select team page with a message 'Your guest account has no channels assigned. Please contact an administrator.' when removed from all the teams and channels.
3. Verify if System message is displayed when Guest User leaves the channel. 

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-18044

### CODE REVIEW

**Comment by prapti** [comment]
@srkgupta MM-18044 keeps hitting this error. Have you seen this? Looks like #teamSidebarWrapper keeps showing up.
Screen Shot 2019-09-26 at 4 51 01 PM

File: e2e/cypress/integration/enterprise/guest_accounts/guest_removal_ui_spec.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    // # Remove the Guest user from all channels of a team as a sysadmin
+    const channels = ['Town Square', 'Off-Topic'];
+    channels.forEach((channel, index) => {
+        // # Remove the Guest User from channel
+        cy.getCurrentChannelId().then((channelId) => {
+            cy.removeUserFromChannel(channelId, guest.id);
+        });
+
+        // * Verify if guest user gets a message when the channel is removed
+        if (index < channels.length - 1 || verifyAlert) {
```
Comment (thekiiingbob) [inline e2e/cypress/integration/enterprise/guest_accounts/guest_removal_ui_spec.js]: This feels a bit confusing to me. Does this modal only appear if it's NOT the last channel? Given that there are just two channels I'd be more inclined to do, albeit a bit nit picky perhaps:

```js
if (channel === 'Town Square' || verifyAlert) {
    // sweet code
}
```

If you don't feel strongly about this, I've approved the PR on my end.

Comment (srkgupta) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_removal_ui_spec.js]: Hi @thekiiingbob 
Yes that's correct. I will simplify the code for better understanding as suggested.

**Comment by srkgupta** [comment]
> 
> 
> @srkgupta MM-18044 keeps hitting this error. Have you seen this? Looks like #teamSidebarWrapper keeps showing up.
> Screen Shot 2019-09-26 at 4 51 01 PM

Hi @prapti 

Upon carefully looking at the screenshot you sent, it was observed that this is actually an issue with the application code and not the Automation test. When a user was removed from the channel, the user is redirected to channel_not_found page instead of redirecting the user to the other team. I have filed a JIRA ticket for tracking this. 

https://mattermost.atlassian.net/browse/MM-19040

Screen Shot 2019-09-29 at 7 18 50 PM

**Comment by prapti** [comment]
@srkgupta Wonderful! Thank you. I'll review your PR quick :)

**Pull request merged**